src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -154,17 +179,54 @@ def fetch_historical_data(
     if not project or not dataset or not alert_rule.organization:
         return None
 
-    historical_data = dataset.timeseries_query(
-        selected_columns=[snuba_query.aggregate],
-        query=snuba_query.query,
-        snuba_params=SnubaParams(
-            organization=alert_rule.organization,
-            projects=[project],
-            start=start,
-            end=end,
-        ),
-        rollup=granularity,
-        referrer=Referrer.ANOMALY_DETECTION_HISTORICAL_DATA_QUERY.value,
-        zerofill_results=True,
-    )
+    if dataset == metrics_performance:
```

Comment:
This was all pulled from https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_sessions.py

Reply:
would be great if we could separate concerns here a bit and pull this into some helpers?

Maybe add a note or create a ticket to do this as part of https://getsentry.atlassian.net/browse/ALRT-187?

Reply:
updated to add `get_crash_free_historical_data()`